### PR TITLE
chore: parallel tests

### PR DESCRIPTION
**Pull request by bharathkkb**
This introduces parallelization when running tests in CI. Sample tests are grouped in to n test groups based [on num projects](https://github.com/terraform-google-modules/terraform-docs-samples/blob/60fbf0ed06e40b5d7e9d3ef3bc0cf069caddf64a/test/setup/main.tf#L18-L21) in setup. I added 4 projects to start with and has resulted in a test time reduction from ~62.5%  (from 4h to 1:30h). 

Test grouping is dynamic and increasing number of projects in the future to reduce time would be a trivial change.

### CODE REVIEW

**Comment by bharathkkb** [comment]
@g-awmalik PTAL

File: build/int.cloudbuild.yaml [hunk 1]
```diff
@@ -21,10 +21,31 @@ steps:
...
+  dir: test/integration
+  waitFor:
+    - prepare
+- id: test-samples-3
+  name: 'gcr.io/cloud-foundation-cicd/$_DOCKER_IMAGE_DEVELOPER_TOOLS:$_DOCKER_TAG_VERSION_DEVELOPER_TOOLS'
+  args: ['/bin/bash', '-c', 'go test -v -timeout 0 -run TestSamples/3']
+  dir: test/integration
+  waitFor:
+    - prepare
+
```
Comment (g-awmalik) [inline build/int.cloudbuild.yaml:48]: nit: whitespace

File: test/integration/sample_test.go [hunk 2]
```diff
@@ -58,20 +73,35 @@ var skipDiscoverDirs = map[string]bool{
...
+	// One test group is associated to one project.
+	groups := []*testGroups{}
+	for i, project := range projects {
+		groups = append(groups, &testGroups{projectID: project, samples: []string{}, group: i})
+	}
+	// Rather than chunking we assign them in a round robin fashion as some samples like sql takes more time.
+	// Chunking would result in all sql* assigned to a single project.
+	// We sort the sample slice beforehand so assignments should be stable for a given run.
+	for i, sample := range samples {
+		idx := i % len(projects)
```
Comment (g-awmalik) [inline test/integration/sample_test.go:103]: nit: rename to groupIndex since i & idx are similar

**Review by g-awmalik — APPROVED** [review]

File: test/integration/sample_test.go (hunk 2 above)
Comment (bharathkkb) [reply test/integration/sample_test.go]: done

**Pull request merged**